internal/stores/gcs.go

```diff
@@ -77,22 +79,10 @@ func (s *GCSStore) Upload(ctx context.Context, objects ...string) error {
 		}
 		defer f.Close()
 
-		// Generate cavorite metadata
-		m, err := metadata.GenerateFromFile(f)
-		if err != nil {
-			return err
-		}
-		logger.V(2).Infof("%s has a checksum of %q", o, m.Checksum)
-		// convert metadata to json
-		blob, err := json.MarshalIndent(m, "", " ")
+		cleanup, err := WriteMetadataToFsys(s, o, f)
```

Comment:
Add a comment on why a function is returned here. We don't use this pattern anywhere else, so calling it out would be helpful when we come back to this in a few weeks/months.